requirements.txt

```diff
@@ -1,113 +1,496 @@
-amqp==2.5.2
-aniso8601==7.0.0
-autopep8==1.4.4
-babel==2.7.0
-beautifulsoup4==4.7.1
-billiard==3.6.1.0
-bleach==2.1.4
-boto3==1.10.41
-botocore==1.13.41
-braintree==3.49.0
-cachetools==3.1.1
-cairocffi==1.1.0
-cairosvg==2.4.2
-celery==4.4.0
-certifi==2019.11.28
-cffi==1.13.2
-chardet==3.0.4
-cssselect2==0.2.2
-defusedxml==0.6.0
-dj-database-url==0.5.0
-dj-email-url==0.2.0
-django==2.2.9
-django-appconf==1.0.3
-django-babel==0.6.2
-django-countries==5.5
-django-filter==2.2.0
-django-graphql-jwt==0.2.1
-django-js-asset==1.2.2
-django-measurement==3.2.0
-django-mptt==0.10.0
-django-phonenumber-field==2.4.0
-django-prices==2.1.0
-django-prices-openexchangerates==1.0.1
-django-prices-vatlayer==1.0.2
-django-render-block==0.6
-django-silk==2.0.0
-django-storages==1.8
-django-templated-email==2.3.0
-django-versatileimagefield==1.11
-docutils==0.15.2
-draftjs-sanitizer==1.0.0
-faker==2.0.5
-freezegun==0.3.12
-google-api-core==1.15.0
-google-auth==1.9.0
-google-cloud-core==1.1.0
-google-cloud-storage==1.23.0
-google-i18n-address==2.3.5
-google-measurement-protocol==1.0.0
-google-resumable-media==0.5.0
-googleapis-common-protos==1.6.0
-gprof2dot==2016.10.13
-graphene==2.1.8
-graphene-django==2.6.0
-graphene-django-optimizer==0.6.1
-graphene-federation==0.0.3
-graphql-core==2.2.1
-graphql-relay==2.0.1
-html-to-draftjs==1.0.1
-html5lib==1.0.1
-idna==2.8
-jinja2==2.10.3
-jmespath==0.9.4
-jsonfield==2.0.2
-kombu==4.6.7
-lxml==4.4.2
-markdown==3.1.1
-markupsafe==1.1.1
-maxminddb==1.5.1
-maxminddb-geolite2==2018.703
-measurement==3.2.0
-mpmath==1.1.0
-oauthlib==3.1.0
-phonenumberslite==8.11.1
-pillow==6.2.1
-prices==1.0.0
-promise==2.3
-protobuf==3.11.1
-psycopg2-binary==2.8.4
-purl==1.5
-pyasn1==0.4.8
-pyasn1-modules==0.2.7
-pycodestyle==2.5.0
-pycparser==2.19
-pygments==2.5.2
-pyjwt==1.7.1
-pyphen==0.9.5
-python-dateutil==2.8.0
-python-magic==0.4.15
-python-magic-bin==0.4.14; sys_platform == "win32"
-pytz==2019.3
-razorpay==1.2.0
-redis==3.3.11
-requests==2.22.0
-rsa==4.0
-rx==1.6.1
-s3transfer==0.2.1
-sentry-sdk==0.13.5
-singledispatch==3.4.0.3
-six==1.13.0
-soupsieve==1.9.5
-sqlparse==0.3.0
-stripe==2.41.0
-sympy==1.5
-text-unidecode==1.3
-tinycss2==1.0.2
-tqdm==4.40.2
-typing==3.7.4.1
-urllib3==1.25.7
-uwsgi==2.0.18; sys_platform != "win32"
-vine==1.3.0
-weasyprint==50
-webencodings==0.5.1
+amqp==2.5.2 \
+    --hash=sha256:6e649ca13a7df3faacdc8bbb280aa9a6602d22fd9d545336077e573a1f4ff3b8 \
```

Comment:
We don't want these hashes: https://docs.getsaleor.com/docs/next/advanced/managing-dependencies/#other-tools

Reply:
Why? It's an important security feature.

Reply:
For some reason, it was breaking our deployment on Heroku. If I remember correctly the problem was that hashes were missing for some requirements and pip failed to install such requirements file.

Reply:
Maybe it's no longer an issue as it works fine with `pip install`. By the way, maybe we should switch our Heroku support to use Docker? https://devcenter.heroku.com/articles/build-docker-images-heroku-yml

Reply:
@IKarbowiak Do you know exactly why it wasn't working? I think it was your PR that removed hashes recently.

Reply:
Yes, it was my PR, but I worked with @fowczarek (because I don't have account on Heroku, and he has one) and he was checking it on Heroku. We saw that poeple writing that it helps. Also I saw that @NyanKiyoshi opened PR to remove hashes, so maybe he can also help here.

Reply:
In my PR I added details on what went wrong:
> Noticed it was not possible to install requirements with some machines as some architecture have wheels available--poetry does not detect the wheel requirements at this time.

But I also saw that somebody said in saleor-docs to use Python 3.8 instead of 3.7. I am pretty sure I was using 3.7, so I would like to try that even though I doubt that–maybe there is no wheel available for 3.8, so it doesn't use them.

For reference, I was installing on the kernel image `4.14.152-127.182.amzn2.x86_64` using `pip install -r requirements.txt` and latest `pip` version that was available.

Once again, 99% sure I was using Python 3.7.4.

Reply:
We've been using Python 3.8.1 on Heroku for some time already and it was failing with this version as well. We were getting this error:

```
Collecting setuptools>=39.2.0 (from cairocffi==1.1.0->-r /tmp/build_2886b32ba14cbeeee7a9d26444bc1fea/requirements.txt (line 34))
       In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
           setuptools>=39.2.0 from https://files.pythonhosted.org/packages/f9/d3/955738b20d3832dfa3cd3d9b07e29a8162edb480bf988332f5e6e48ca444/setuptools-44.0.0-py2.py3-none-any.whl#sha256=180081a244d0888b0065e18206950d603f6550721bd6f8c0a10221ed467dd78e (from cairocffi==1.1.0->-r /tmp/build_2886b32ba14cbeeee7a9d26444bc1fea/requirements.txt (line 34))
 !     Push rejected, failed to compile Python app.
```

Reply:
Also, `setuptools` is not specified in our requirements files. Others seem to have this issue as well: https://github.com/python-poetry/poetry/issues/1584

Reply:
Wait, why are we running with `--require-hashes`?

Reply:
I can't see `--require-hashes` anywhere in our code, not sure where it comes from.

Reply:
It comes by default when you have hashes in the file. There is no flag to tell it to act differently from the requirements of wheels. I believe the issue is that poetry doesn't/is not able to detect the requirements from wheels thus generate the requirement file with hashes for it as the wheels can differ from platform to platform.

Reply:
Why do you think it's related to wheels? The only issue I've seen in this thread is related to missing hashed for the three blacklisted packages (setuptools, pip, distribute).

Reply:
I recall having troubles with other dependencies than setuptools. But I'm not sure anymore, I would need to check if that was the only failure.

Reply:
@kswiatek92 Please see this comment, maybe it will help: https://github.com/mirumee/saleor/pull/5109#issuecomment-575041918

Reply:
@marcin
it doesn't help
it brings different problems
when removing `sys_platform` from `python-magic-bin` it returns
```
No matching distribution found for python-magic-bin==0.4.14 (from -r /tmp/build_5ee20a5feba89390713ce1aa800698d1/requirements.txt (line 421))
```

Reply:
removed hashes and created bug ticket for it